ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs

```diff
@@ -90,368 +65,246 @@ impl Worker {
-                        Some(size_in_bytes),
-                        Some((last_transaction.version - first_transaction.version + 1) as i64),
-                        None,
+                None => {
+                    tracing::warn!(
+                        "[Indexer Cache] File store metadata not found. Waiting for {} ms.",
+                        FILE_STORE_METADATA_WAIT_MS
                     );
-                },
-                Err(e) => {
-                    ERROR_COUNT
-                        .with_label_values(&["failed_to_update_cache_version"])
-                        .inc();
-                    bail!("Update cache with version failed: {}", e);
-                },
-            }
-            Ok(GrpcDataStatus::ChunkDataOk {
-                start_version,
-                num_of_transactions: transaction_len as u64,
-            })
-        },
-    }
-}
-
-//// Setup the cache operator with init signal, includeing chain id and starting version from fullnode.
-async fn setup_cache_with_init_signal(
-    conn: redis::aio::ConnectionManager,
-    init_signal: TransactionsFromNodeResponse,
-) -> Result<(
-    CacheOperator<redis::aio::ConnectionManager>,
-    ChainID,
-    StartingVersion,
-)> {
-    let (fullnode_chain_id, starting_version) = match init_signal
-        .response
-        .expect("[Indexer Cache] Response type does not exist.")
-    {
-        Response::Status(status_frame) => {
-            match StatusType::try_from(status_frame.r#type)
-                .expect("[Indexer Cache] Invalid status type.")
-            {
-                StatusType::Init => (init_signal.chain_id, status_frame.start_version),
-                _ => {
-                    bail!("[Indexer Cache] Streaming error: first frame is not INIT signal.");
+                    tokio::time::sleep(std::time::Duration::from_millis(
+                        FILE_STORE_METADATA_WAIT_MS,
+                    ))
+                    .await;
                 },
             }
-        },
-        _ => {
-            bail!("[Indexer Cache] Streaming error: first frame is not siganl frame.");
-        },
-    };
+        };
+        let starting_version = file_store_metadata.version;
 
-    let mut cache_operator = CacheOperator::new(conn);
-    cache_operator.cache_setup_if_needed().await?;
-    cache_operator
-        .update_or_verify_chain_id(fullnode_chain_id as u64)
-        .await
-        .context("[Indexer Cache] Chain id mismatch between cache and fullnode.")?;
+        // Check cache chain id.
+        let cache_chain_id = cache_operator
+            .get_chain_id()
+            .await?
+            .context("Failed to get chain id from cache")?;
+        ensure!(cache_chain_id == file_store_metadata.chain_id);
 
-    Ok((cache_operator, fullnode_chain_id, starting_version))
-}
+        // TODO: cache worker should restart from file store version.s
+        // For now, let's pick the max of cache.starting_version and file_store_metadata
```

Comment:
picking the max is fine. because we're assuming everything from starting_version ... cache_latest_version is already in the cache? can you explain why in the comment.